src/uu/rm/src/rm.rs

```diff
@@ -333,6 +346,83 @@ pub fn uu_app() -> Command {
         )
 }
 
+/// Creates a progress bar for rm operations if conditions are met.
+/// Returns Some(ProgressBar) if `total_files` > 0, None otherwise.
+fn create_progress_bar(files: &[&OsStr], recursive: bool) -> Option<ProgressBar> {
+    if let Ok(total_files) = count_files(files, recursive) {
```

Comment:
I would use a traditional assignment here to get rid of one indentation level. Something like:
```rust
    let total_files = count_files(files, recursive).ok()?;
```